eclipse-ee4j/jaxrs-api
# TCK Migration: Move pending tests from jakartaee-tck/src/com/sun/ts/tests/jaxrs/api (#1015)

This PR migrates all tests from `jakartaee-tck/src/com/sun/ts/tests/jaxrs/api` in https://github.com/eclipse-ee4j/jakartaee-tck.

Linked issues: #1015 

To test the changes :

```
export JAVA_HOME=<JDK11_HOME>
export M2_HOME=<MAVEN_HOME>
export PATH=$M2_HOME/bin:$JAVA_HOME/bin:$PATH
```

In `jaxrs-tck/`:
`mvn clean install`

In `jersey-tck/`:
run all tests : `mvn clean verify -Parq-glassfish-managed`

**Test failures:** (those should be tracked in #1020)

* `api/rs/core/newcookie`: `equalsTest` fails because `NewCookie.hashcode()` throws a `NullPointerException` because sameSite is `null` (test disabled)
* `api/rs/core/linkjaxbadapter`: Implementation for `jakarta.xml.bind-api` cannot be found and I don't know how to add it (tests disabled)
* `api/rs/ext/interceptor`: Tests fail with `IllegalStateException: [FATAL] HttpRequest is null` (tests disabled)
